forum/investors/views.py

```diff
@@ -1,3 +1,57 @@
-from django.shortcuts import render
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status
+from django.shortcuts import get_list_or_404, get_object_or_404
 
-# Create your views here.
+from .models import Investor
+from .serializers import InvestorSerializer
+
+from users.permissions import *
+from rest_framework.permissions import IsAuthenticated
+
+
+class UserInvestorsView(APIView):
+    permission_classes = [
+        IsAuthenticated,
+        ThisUserPermission,
+        IsInvestorNamespaceSelected
+    ]
+
+    def get(self, request, user_id):
+        investors = get_list_or_404(Investor, user=user_id)
+        serializer = InvestorSerializer(investors, many=True)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    def post(self, request, user_id):
+        serializer = InvestorSerializer(data={**request.data, 'user': user_id})
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class UserInvestorView(APIView):
+    permission_classes = [
+        IsAuthenticated,
+        ThisUserPermission,
+        IsInvestorNamespaceSelected,
+        ThisInvestor
+    ]
+
+    def get(self, request, user_id, investor_id):
+        investor = get_object_or_404(Investor, user=user_id, investor_id=investor_id)
+        serializer = InvestorSerializer(investor)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    def patch(self, request, user_id, investor_id):
+        investor = get_object_or_404(Investor, user=user_id, investor_id=investor_id)
+        serializer = InvestorSerializer(investor, data=request.data, partial=True)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_200_OK)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    def delete(self, request, user_id, investor_id):
+        investor = get_object_or_404(Investor, user=user_id, investor_id=investor_id)
+        investor.delete()
```

Comment:
We agreed to not delete rows from database in startups. We mark them as deleted instead. Maybe we can use the same approach here

Reply:
Yes, of course. I have changed it. Now, on delete, investor is marked as deleted instead of real deleting.